res/controllers/Reloop-Mixage.scripts.js

```diff
@@ -0,0 +1,812 @@
+            Mixage.startLoopAdjust(group);
+        }
+    } else {
+        if (value === DOWN) { // loop adjustment mode is not active
+            engine.setValue(group, "reloop_toggle", 1);
+        } else {
+            engine.setValue(group, "reloop_toggle", 0);
+        }
+    }
+};
+
+// set a loop in point if none is defined, otherwise enable adjustment of the start position with the jogwheel
+Mixage.handleLoopIn = function(_channel, _control, value, _status, group) {
+    if (Mixage.adjustLoop[group]) { // loop adjustment mode is active
+        if (Mixage.adjustLoopOut[group] && value === DOWN) { // loop out is currently being adjusted, switch to loop in
+            Mixage.startLoopAdjust(group, "start");
+        } else if (Mixage.adjustLoopIn[group] && value === DOWN) { // loop in is currently being adjusted switch to loop in and out
+            Mixage.startLoopAdjust(group);
+        }
+    } else { // loop adjustment mode is not active
+        if (value === DOWN) {
+            engine.setValue(group, "loop_in", 1);
+        } else {
+            engine.setValue(group, "loop_in", 0);
+        }
+    }
+};
+
+// set a loop in point if none is defined, otherwise enable adjustment of the start position with the jogwheel
+Mixage.handleLoopOut = function(_channel, _control, value, _status, group) {
+    if (Mixage.adjustLoop[group]) { // loop adjustment mode is active
+        if (Mixage.adjustLoopIn[group] && value === DOWN) { // loop in is currently being adjusted, switch to loop out
+            Mixage.startLoopAdjust(group, "end");
+        } else if (Mixage.adjustLoopOut[group] && value === DOWN) { // loop out is currently being adjusted switch to loop in and out
+            Mixage.startLoopAdjust(group);
+        }
+    } else {
+        if (value === DOWN) { // loop adjustment mode is not active
+            engine.setValue(group, "loop_out", 1);
+        } else {
+            engine.setValue(group, "loop_out", 0);
+        }
+    }
+};
+
+// Toggle play and make sure the preview deck stops when starting to play in a deck
+// brake or softStart a while the scratch toggle button is held
+Mixage.handlePlay = function(_channel, _control, value, _status, group) {
+    const deck = script.deckFromGroup(group);
+    if (value === DOWN && Mixage.scratchPressed[group]) { // scratch toggle is pressed
+        if (engine.getValue(group, "play") === 0) {
+            engine.softStart(deck, true, 1.5);
+        } else {
+            engine.brake(deck, true, 0.75);
+        }
+    } else if (value === DOWN) { // scratch toggle is not pressed
+        script.toggleControl(group, "play");
+    }
+};
+
+// Checks whether the Traxx button is double pressed
+Mixage.handleTraxPress = function(channel, control, value, status, group) {
+    if (value === DOWN) {
+        if (Mixage.traxxPressTimer === 0) { // first press
+            Mixage.traxxPressTimer = engine.beginTimer(400, function() {
```

Comment:
I suggest to increase the timeout a bit, just to improve accessibility for all usesr that are not able to press buttons quickly (like they can adjust the mouse double-click timeout on system level). 1000 is safe IMO

Reply:
I increased the timeout for the double press on the traxx button to 800 ms. I also added the `Mixage.shortPressTimeout` and `Mixage.doublePressTimeout` so these timeouts can be adjusted by users more easily